### PR TITLE
Operator Improvements

### PR DESCRIPTION
**Pull request by BLasan**
## Purpose
It's better to add separate secrets for pull and push scenarios. (Allow users to define pull secret names rather than hard coding it)

## Goals
Users can now define array of imagePullSecretNames in docker_registry_conf.yaml.

## Approach
- Add new field "imagePullSecretName" under` docker-registry-config` data section.
- Check the presence of this value. If it's empty then use the default secret name used for pushing, or else use the array of pull secret names for the configurations.

## Documentation
Needs to be updated.

## Related PRs
#487

### CODE REVIEW

File: api-operator/deploy/controller-configs/docker_registry_conf.yaml [hunk 1]
```diff
@@ -23,4 +23,6 @@ data:
   #docker registry type which the mgw image to be pushed. supported types: DOCKER_HUB, AMAZON_ECR, GCR, HTTP.   Default->  registryType: DOCKER_HUB
   registryType: DOCKER_HUB
   #docker repository name which the mgw image to be pushed.  eg->  repositoryName: docker.io/{USER_NAME of Docker Hub account}
-  repositoryName: REPOSITORY_NAME_OF_DOCKER_REGISTRY
\ No newline at end of file
+  repositoryName: DOCKER_REPOSITORY_NAME
+  #pull secret array
+  imagePullSecretName: docker-my-secret-pull
```
Comment (renuka-fernando) [inline api-operator/deploy/controller-configs/docker_registry_conf.yaml:28]: shall we make the default value empty

Comment (renuka-fernando) [reply api-operator/deploy/controller-configs/docker_registry_conf.yaml:28]: and let's add a new line

Comment (BLasan) [reply api-operator/deploy/controller-configs/docker_registry_conf.yaml:28]: > shall we make the default value empty

sure :)

**Comment by BLasan** [comment]
@renuka-fernando Changes added. Please review

File: api-operator/deploy/controller-configs/docker_registry_conf.yaml (hunk 1 above)
Comment (renuka-fernando) [reply api-operator/deploy/controller-configs/docker_registry_conf.yaml]: imagePullSecretName -> imagePullSecretNames

Comment (BLasan) [reply api-operator/deploy/controller-configs/docker_registry_conf.yaml]: Updated

File: api-operator/deploy/controller-configs/docker_registry_conf.yaml [hunk 2]
```diff
@@ -23,4 +23,6 @@ data:
   #docker registry type which the mgw image to be pushed. supported types: DOCKER_HUB, AMAZON_ECR, GCR, HTTP.   Default->  registryType: DOCKER_HUB
   registryType: DOCKER_HUB
   #docker repository name which the mgw image to be pushed.  eg->  repositoryName: docker.io/{USER_NAME of Docker Hub account}
-  repositoryName: REPOSITORY_NAME_OF_DOCKER_REGISTRY
\ No newline at end of file
+  repositoryName: DOCKER_REPOSITORY_NAME
+  #pull secret array
+  imagePullSecretNames:
```
Comment (pubudu538) [inline api-operator/deploy/controller-configs/docker_registry_conf.yaml:28]: Add sample values

**Comment by BLasan** [comment]
@pubudu538 Added sample values for imagePullSecretNames

File: api-operator/pkg/registry/utils/utils.go [hunk 3]
```diff
@@ -59,3 +61,24 @@ func IsImageExists(auth RegAuth, imageRepository string, imageName string, tag s
 	}
 	return false, nil
 }
+
+func GetPullSecrets(pullSecretDef string, pullSecretName string) []corev1.LocalObjectReference {
+	// var pullSecretDef = &PullSecretDefined
+	// var pushSecretDef = &registry.PushSecretDefned
+	var pullSecretArray = []corev1.LocalObjectReference{}
```
Comment (pubudu538) [inline api-operator/pkg/registry/utils/utils.go:68]: Shall we remove this commented code?

Comment (BLasan) [reply api-operator/pkg/registry/utils/utils.go:66]: Removed. Please review

**Review by pubudu538 — APPROVED** [review]

File: api-operator/pkg/registry/utils/utils.go (hunk 3 above)
Comment (BLasan) [reply api-operator/pkg/registry/utils/utils.go:66]: Fixed via [f1b3ac5](https://github.com/wso2/k8s-api-operator/pull/555/commits/f1b3ac5d7418d9479b19828218edb9cb7a8b7001)

File: api-operator/deploy/controller-configs/docker_registry_conf.yaml (hunk 1 above)
Comment (BLasan) [reply api-operator/deploy/controller-configs/docker_registry_conf.yaml]: Fixed via [13add0d](https://github.com/wso2/k8s-api-operator/pull/555/commits/13add0dcfd3e391e2ab922d11a978e285223f363)

**Pull request merged**